### DISCUSSION TITLE
How to add a third-party custom element?

### DISCUSSION THREAD

**Discussion by jeremeevans** [nuxt/nuxt#18196] [Questions] [answered]
In Nuxt v2, I was able to add third party custom element globally using plugins in the nuxt config:

  plugins: [
    {src: "~node_modules/path-to-component", ssr: false },
  ],

and then use it in my templates. What's the correct way to do that in v3?

For reference, the elements I am looking to import were built using LitElement, and they register themselves as custom elements using customElements.define...

**Comment by wobsoriano** [comment]
Related https://github.com/nuxt/framework/discussions/1214

**Comment by dammy001** [comment]
import { createVuetify } from 'vuetify'

export default defineNuxtPlugin((nuxtApp) => {
   const vuetify = createVuetify(...) // Replaces new Vuetify(...)
   nuxtApp.vueApp.use(vuetify)
});

**Reply by wobsoriano** [reply]
He's asking about how to ignore custom elements and not register plugins.

https://v3.vuejs.org/guide/migration/custom-elements-interop.html

**Reply by dammy001** [reply]
Oh that's true. Thanks man

**Reply by jeremeevans** [reply]
Where in the Nuxt 3 configuration can I access the `compilerOptions`?

_**EDIT: Answered below by @danielroe **_

**Comment by danielroe** [comment, marked as answer]
You can do it like this:
```js
import { defineNuxtConfig } from 'nuxt3'

export default defineNuxtConfig({
  build: {
    transpile: ['vaadin']
  },
  vue: {
    compilerOptions: {
      isCustomElement: tag => ['vaadin-checkbox'].includes(tag)
    }
  }
})
```
**`~/plugins/components.client.ts`**
```js
import '@vaadin/vaadin-checkbox'
export default () => {}
```

**Reply by jeremeevans** [reply]
It doesn't look like compilterOptions is defined in the Nuxt config. I'm trying to find them... somewhere?? Not sure how to do it.
![image](https://user-images.githubusercontent.com/1747006/138900588-f2ee641e-e819-4205-8bd5-c6122a8f559b.png)

**Reply by danielroe** [reply]
Ensure you're on the latest Nuxt version: just merged in https://github.com/nuxt/framework/pull/1452...

**Reply by jeremeevans** [reply]
Awesome, thank you! That worked.

**Comment by eMeRiKa13** [comment]
Hi,

I'm new to Nuxt, directly starting with the version 3 (why not!). 

I'm trying to use this Vue3 component https://github.com/SortableJS/vue.draggable.next and found this discussion, how to add a third-party custom element.

So I created a `plugin/components.client.ts` file with

```
import { defineNuxtConfig } from 'nuxt3'

export default defineNuxtConfig({
  build: {
    transpile: ['draggable']
  },
  vue: {
    compilerOptions: {
      isCustomElement: tag => ['draggable'].includes(tag)
    }
  }
})
```

And then in my page I use it (without adding nothing in the script setup):
```
<draggable
    tag="ul"
    :list="list"
    handle=".handle"
    item-key="id"
>
    <template #item="{ element, index }">
      <li>               
         ...
      </li> 
    </template>
</draggable>
```

But it's not working, and the console displays

> [Vue warn]: Failed to resolve component: draggable
If this is a native custom element, make sure to exclude it from component resolution via compilerOptions.isCustomElement. 

What I am missing? Thank you

**Reply by jeremeevans** [reply]
The Nuxt config part should go in nuxt.config.ts ... the components.client.ts should import the draggable component and export default, so all it should have is something like this:

```
import "draggable"
export default () => {}
```

**Reply by eMeRiKa13** [reply]
ho okay I see, I read this discussion but did it complementally wrong.

thank you for your answer
